src/gsy_e/models/market/forward.py

```diff
@@ -86,6 +130,14 @@ def _get_end_time(current_time: DateTime) -> DateTime:
     def _get_market_slot_duration(_current_time: DateTime, _config) -> duration:
         return duration(minutes=15)
 
+    @property
+    def _time_from_market_close_till_delivery(self) -> duration:
+        return duration(minutes=15)
```

Comment:
I have the feeling that this indirectly replicates the logic of `_get_end_time()`.

Reply:
I do not think so. The _get_end_time sets the time horizon over which new markets will be created. However the closing time of these markets is implicitly defined by the ForwardMarketRotator (which I am also in the process of updating) , and more specifically by its abstract method `_get_last_slot_to_be_deleted`. This is set for the yearly markets to this `current_time.add(years=1)` which is correct for the yearly market (essentially we want to close all markets before 2023, for the example that you presented above).